src/libcore/convert.rs

```diff
@@ -554,6 +554,17 @@ impl<T> From<T> for T {
     fn from(t: T) -> T { t }
 }
 
+#[stable(feature = "convert_infallible", since = "1.34.0")]
+#[cfg(not(boostrap_stdarch_ignore_this))]
+#[rustc_reservation_impl="a future version of Rust might implement `From<!>` for \
```

Comment:
One other request: Can you cite the tracking issue https://github.com/rust-lang/rust/issues/64631 here?